art_internal_impl.hpp

```diff
@@ -1059,24 +1347,89 @@ class basic_inode_4 : public basic_inode_4_parent<ArtPolicy> {
 #endif  // #ifdef UNODB_DETAIL_X86_64
   }
   UNODB_DETAIL_RESTORE_MSVC_WARNINGS()
-
-  constexpr void delete_subtree(db &db_instance) noexcept {
+      
+  [[nodiscard]] constexpr node_ptr get_child(const std::uint8_t child_index) noexcept {
```

Comment:
Here and everywhere: no `const` on pass-by-value parameters

Reply:
Ok.  I am actually in favor of const in these situations because otherwise it is not possible to give the argument const semantics in the implementation.  If you love const, use it everywhere.  The method can always make a local copy if it needs to modify the passed argument.

I will make the change.